### PR TITLE
feat: support beforeClose on other env

### PR DESCRIPTION
**Pull request by popomore**
##### Checklist

- [ ] `npm test` passes
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message follows commit guidelines

##### Description of change

Closes https://github.com/eggjs/egg/issues/1267

Closes https://github.com/eggjs/egg-cluster/pull/47

### CODE REVIEW

**Comment by popomore** [comment]
先做个非优雅的版本

File: lib/agent_worker.js [hunk 1]
```diff
@@ -38,4 +38,5 @@ agent.once('error', startErrorHandler);
 gracefulExit({
   logger: consoleLogger,
   label: 'agent_worker',
+  beforeExit: () => agent.close(),
```
Comment (fengmk2) [inline lib/agent_worker.js]: beforeExit 得支持一般方法才行

Comment (popomore) [reply lib/agent_worker.js]: 还在改

**Comment by popomore** [comment]
https://github.com/node-modules/graceful-process/pull/4

**Comment by fengmk2** [comment]
还是不行？

**Comment by popomore** [comment]
ci 下不太稳定

**Comment by popomore** [comment]
已经过了

**Comment by popomore** [comment]
合并发布

**Comment by popomore** [comment]
+ egg-cluster@1.15.0

**Pull request merged**